File: haohaninfo/交易實務案例/94.R
#取得即時報價，詳細在技巧51
source("function.R")

#假設目前開倉並且開倉價位在10000
index <- 1
orderPrice <- 10000

#設定變數初始格式
inDesk<-0
outDesk<-0

while(index!=0){
 #取得即時成交資訊
 Mdata<-GetMatchData(DataPath,Date)
 MatchTime <- strptime(Mdata[[1]][1],'%H:%M:%OS')
 MatchPrice <- as.numeric(Mdata[[1]][2])
 MatchQty <- as.numeric(Mdata[[1]][3])
 
 #取得上下五檔價資訊
 UpDn5data<-GetUpDn5Data(DataPath,Date) 
 Dn1Price<- as.numeric(UpDn5data[[1]][2])
 Up1Price<- as.numeric(UpDn5data[[1]][12])

 #進行內外盤判斷
 if ( MatchPrice >= Up1Price ){
  outDesk <- outDesk+MatchQty
 }else if( MatchPrice <= Dn1Price){
  inDesk <- inDesk+MatchQty
 }

 #判斷內外盤量反轉
 if(index ==1 ){
  if(outDesk < inDesk){
   index <- 0
   print("Cover Buy Success!")
   break
  }
 }else if(index==(-1)){
  if(outDesk > inDesk){
   index <- 0
   print("Cover Sell Success!")
   break
  }
 }


}

File: haohaninfo/即時範例/94.py
# -*- coding: UTF-8 -*-

#取得報價資訊，詳情請查看技巧51
execfile('function.py')

#定義指標變數
OutDesk=0
InDesk=0

#假設目前倉位為買方，進場部分請參考第七章
index=1
orderPrice=10300
coverPrice=0

#取得成交資訊
for i in getMatch():		
 MatchInfo=i.split(',')
 MatchTime=datetime.datetime.strptime(MatchInfo[0],"%H:%M:%S.%f")
 MatchPrcie=int(MatchInfo[1])
 MatchQty=int(MatchInfo[2])
 UpDn5Info=getLastUpDn5()
 Dn1Price=int(UpDn5Info[1])
 Up1Price=int(UpDn5Info[11])

 if MatchPrcie>=Up1Price:
  OutDesk+=MatchQty
 if MatchPrcie<=Dn1Price:
  InDesk+=MatchQty
 
 #出場判斷
 if index==1:
  if InDesk>OutDesk:
   index=0
   coverPrice=MatchPrice
   print MatchInfo[0],"Order Sell Success!"
   break
 elif index==-1:
  if InDesk<OutDesk: 
   index=0
   coverPrice=MatchPrice
   print MatchInfo[0],"Order Buy Success!"
   break
 
 print MatchInfo[0],"OutDesk",OutDesk,"InDesk",InDesk


